### PR TITLE
Suppress some poet deprecations

### PR DESCRIPTION
**Pull request by ZacSweers**
These are unlikely to actually go away in 2.0 since they make sense in this case

### CODE REVIEW

**Review by Egorand — APPROVED** [review]
Would it be helpful to add a short comment to each suppress to explain why it's being suppressed?

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt [hunk 1]
```diff
@@ -231,6 +231,7 @@ internal fun targetType(
   val resolvedTypes = mutableListOf<ResolvedTypeMapping>()
   val superTypes = appliedType.supertypes(types)
     .filterNot { supertype ->
+      @Suppress("DEPRECATION")
```
Comment (ZacSweers) [inline kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt:234]:
```suggestion
      @Suppress("DEPRECATION") // Appropriate in this case
```

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt [hunk 2]
```diff
@@ -270,6 +271,7 @@ internal fun targetType(
         val superSuperClass = supertype.element.superclass as DeclaredType
 
         // Convert to an element and back to wipe the typed generics off of this
+        @Suppress("DEPRECATION")
```
Comment (ZacSweers) [inline kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt:274]:
```suggestion
        @Suppress("DEPRECATION") // Appropriate in this case
```

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt [hunk 3]
```diff
@@ -285,6 +287,7 @@ internal fun targetType(
     }
 
   for ((localAppliedType, supertypeApi) in superTypes.entries) {
+    @Suppress("DEPRECATION")
```
Comment (ZacSweers) [inline kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt:290]:
```suggestion
    @Suppress("DEPRECATION") // Appropriate in this case
```

File: kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt [hunk 4]
```diff
@@ -311,6 +314,8 @@ internal fun targetType(
       .any { it.isInternal }
     if (forceInternal) KModifier.INTERNAL else visibility
   }
+
+  @Suppress("DEPRECATION")
```
Comment (ZacSweers) [inline kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt:318]:
```suggestion
  @Suppress("DEPRECATION") // Appropriate in this case
```

**Pull request merged**